### PR TITLE
add gnll loss

### PR DESCRIPTION
**Pull request by Justinezgh**
I add the Gaussian negative log likelihood loss (from https://arxiv.org/pdf/1906.03156.pdf) and changed the train_compressor script accordingly

### CODE REVIEW

File: scripts/train_compressor.py [hunk 1]
```diff
@@ -101,15 +100,11 @@ def __call__(self, y):
...
-    print("ResNet34")
-
-    compressor = hk.transform_with_state(lambda y: ResNet34(dim)(y, is_training=True))
-
-elif args.resnet == "resnet18":
-    print("ResNet18")
-
+if args.loss == "train_compressor_gnll":
+    compressor = hk.transform_with_state(
+        lambda y: ResNet18(int(dim + ((dim**2) - dim) / 2 + dim))(y, is_training=True)
```
Comment (aboucaud) [inline scripts/train_compressor.py:105]: The only difference in your if statement is the input dimension of the ResNet18. Then it should look like
```python 
if args.loss == "train_compressor_gnll":
    # Comment on how the dim is computed
    resnet_dim = int(dim + ((dim**2) - dim) / 2 + dim)
else:
    resnet_dim = dim
compressor = hk.transform_with_state(lambda y: ResNet18(resnet_dim)(y, is_training=True))

File: scripts/train_compressor.py [hunk 2]
```diff
@@ -127,12 +122,21 @@ def __call__(self, y):
     parameters_compressor = hk.data_structures.merge(parameters_resnet, params_nf)
 elif args.loss == "train_compressor_mse":
     parameters_compressor = parameters_resnet
+elif args.loss == "train_compressor_gnll":
+    parameters_compressor = parameters_resnet
```
Comment (aboucaud) [inline scripts/train_compressor.py:126]: Could be simplified to `args.loss in ["train_compressor_mse", "train_compressor_gnll"]`

File: scripts/train_compressor.py [hunk 3]
```diff
@@ -127,12 +122,21 @@ def __call__(self, y):
...
 
 
 # define optimizer
 total_steps = args.total_steps
+
+if args.loss == "train_compressor_gnll":
+    start_lr = 0.0001
+
+else:
+    start_lr = 0.001
```
Comment (aboucaud) [inline scripts/train_compressor.py:136]: You could explicitely define at the top of the training script a 
`DEFAULT_LEARNING_RATE = 0.001` constant value so this value is not hidden at the bottom of the script.

Same for the BATCH_SIZE for instance, even if you don't modify it.

**Review by aboucaud — COMMENTED** [review]
Review mostly about Pythonic stuff.
